pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/ValidatorTest.java

```diff
@@ -484,6 +484,14 @@ public void notificationsSectionRemoved() throws Exception {
                 .go();
     }
 
+    @Issue("JENKINS-41645")
+    @Test
+    public void invalidEnvironmentIdentifiers() throws Exception {
+        expectError("invalidEnvironmentIdentifiers")
+                .logContains(Messages.ModelParser_InvalidEnvironmentIdentifier("1FOO = \"BLARGH\""))
```

Comment:
Valid point. The other formats mentioned in the JIRA barf out before we get to validation, so I didn't think to add them. Lemme consider that.